pkg/server/server.go

```diff
@@ -451,30 +449,21 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 
 	// below this point, don't throw errors or we may leak resources in tests
 
-	cycleDetectionCheckResolver := graph.NewCycleDetectionCheckResolver()
-	s.checkResolver = cycleDetectionCheckResolver
-
-	localChecker := graph.NewLocalChecker(
+	var checkBuilderOpts []graph.CheckQueryBuilderOpt
```

Comment:
```suggestion
	checkBuilderOpts := []graph.CheckQueryBuilderOpt{
	    graph.WithLocalCheckerOpts(
	      graph.WithResolveNodeBreadthLimit(s.resolveNodeBreadthLimit),
	    ),
	}
```
Just initialize the variable with the initial value instead of the immediate append afterwards.